gateway/src/apicast/policy/metrics/metrics.lua

```diff
@@ -0,0 +1,113 @@
+local _M = require('apicast.policy').new('Metrics')
+
+local resty_env = require('resty.env')
+local errlog = require('ngx.errlog')
+local prometheus = require('apicast.prometheus')
+local metrics_updater = require('apicast.metrics.updater')
+local tonumber = tonumber
+local select = select
+local find = string.find
+local pairs = pairs
+
+local new = _M.new
+
+local log_map = {
+  'emerg',
+  'alert',
+  'crit',
+  'error',
+  'warn',
+  'notice',
+  'info',
+  'debug',
+}
+
+local log_level_env = 'METRICS_LOG_LEVEL'
+local max_logs_env = 'METRICS_MAX_LOGS'
+
+local log_level_default = 'error'
+local max_logs_default = 100
+
+local function find_i(t, value)
```

Comment:
Now when thinking about it. It might be good to hash this as `value => i` so we have constant time lookup.